registration-processor/registration-processor-message-sender-impl/src/main/java/io/mosip/registration/processor/message/sender/service/impl/MessageNotificationServiceImpl.java

```diff
@@ -535,10 +537,12 @@ private Map<String, Object> setAttributes(String idJsonString, Map<String, Objec
 
 			demographicIdentity = JsonUtil.objectMapperReadValue(idJsonString, JSONObject.class);
 
-
-		String mapperJsonString = Utilities.getJson(utility.getConfigServerFileStorageURL(),
-				utility.getGetRegProcessorIdentityJson());
-		JSONObject mapperJson = JsonUtil.objectMapperReadValue(mapperJsonString, JSONObject.class);
+        if(mapperJson==null) {
+        	String mapperJsonString = Utilities.getJson(utility.getConfigServerFileStorageURL(),
+    				utility.getGetRegProcessorIdentityJson());
+    		 mapperJson = JsonUtil.objectMapperReadValue(mapperJsonString, JSONObject.class);
+        }
+	
 		JSONObject mapperIdentity = JsonUtil.getJSONObject(mapperJson, utility.getGetRegProcessorDemographicIdentity());
 
 		List<String> mapperJsonKeys = new ArrayList<>(mapperIdentity.keySet());

```

Comment:
can we rather cache this to avoid ceation of arraylist everytime ?

Reply:
I have cached both and fix the issue because mapperIdentity  being used in next for loop.
mapperJsonKeys and mapperIdentity